packages/notifi-react-card/lib/hooks/useNotificationHistory.tsx

```diff
@@ -0,0 +1,149 @@
+      (notification) => {
+        const detail =
+          notification.detail as HealthValueOverThresholdEventDetails;
+        const threshold = detail.threshold ?? '';
+        const name = detail.name ?? '';
+        const value = detail.value ?? '';
+        let thresholdDirection = '';
+        if (parseFloat(value) > parseFloat(threshold)) {
+          thresholdDirection = 'under';
+        } else {
+          thresholdDirection = 'over';
+        }
+        return {
+          notificationTitle: 'Health Check',
+          notificationImage: <RatioCheckIcon />,
+          notificationSubject: `${name} ${thresholdDirection} ${threshold}`,
+          notificationDate: notification.createdDate,
+          notificationMessage: undefined,
+        };
+      },
+    ),
+    new SupportedEntryDetail('GenericEventDetails', (notification) => {
+      const detail = notification.detail as GenericEventDetails;
+      return {
+        notificationTitle: detail.sourceName,
+        notificationImage: <AlertIcon icon={detail.icon} />,
+        notificationSubject: detail.notificationTypeName,
+        notificationDate: notification.createdDate,
+        notificationMessage: detail.genericMessage,
+      };
+    }),
+    new SupportedEntryDetail(
+      'ChatMessageReceivedEventDetails',
+      (notification) => {
+        const detail = notification.detail as ChatMessageReceivedEventDetails;
+        return {
+          notificationTitle: `New Message from ${detail.senderName}`,
+          notificationSubject: `New Message from ${detail.senderName}`,
+          notificationDate: notification.createdDate,
+          notificationMessage: detail.messageBody,
+          notificationImage: <ChatAlertIcon width={17} height={17} />,
+        };
+      },
+    ),
+    new SupportedEntryDetail(
+      'AccountBalanceChangedEventDetails',
+      (notification) => {
+        const detail = notification.detail as AccountBalanceChangedEventDetails;
+        const changeAmount = formatAmount(
+          Math.abs(detail.previousValue - detail.newValue),
+        );
+        const getTitle = () => {
+          return detail.direction === 'INCOMING'
+            ? `Incoming Transaction: ${changeAmount} ${detail.tokenSymbol}`
+            : `Outgoing Transaction: ${changeAmount} ${detail.tokenSymbol}`;
+        };
+        const walletAddress = notification.sourceAddress ?? '';
+        return {
+          notificationImage: <SwapIcon />,
+          notificationTitle: getTitle(),
+          notificationSubject: getTitle(),
+          notificationDate: notification.createdDate,
+          notificationMessage: `Wallet ${walletAddress} account balance changed by ${changeAmount} ${detail.tokenSymbol}`,
+        };
+      },
+    ),
+  ];
+
+  const validateIsSupported = (entry?: NotificationHistoryEntry): boolean => {
+    if (
+      supportedHistoryDetails
+        .map((d) => d.typename)
+        .includes(entry?.detail?.__typename ?? '')
```

Comment:
the function `validateIsSupported` is run many times a frame -- let's consider improving its performance so we don't:
1. create a new list via .map
2. iterate over it (O[n]) via .includes

since the number of supportedHistoryDetails is fairly low, I think it's sufficient to construct a Set<string> beforehand (e.g. at the module level)